x/concentrated-liquidity/internal/math/tick.go

```diff
@@ -1,52 +1,165 @@
 package math
 
 import (
+	"fmt"
+
 	sdk "github.com/cosmos/cosmos-sdk/types"
 
-	"github.com/osmosis-labs/osmosis/osmomath"
+	"github.com/osmosis-labs/osmosis/v13/x/concentrated-liquidity/types"
 )
 
-var TickBase = osmomath.MustNewDecFromStr("1.0001")
+var sdkNineDec = sdk.NewDec(9)
+var sdkTenDec = sdk.NewDec(10)
 
-// TicksToSqrtPrice returns the sqrt price for the lower and upper ticks.
-// Returns error if fails to calculate sqrt price.
+// TicksToPrice returns the price for the lower and upper ticks.
+// Returns error if fails to calculate price.
 // TODO: spec and tests
-func TicksToSqrtPrice(lowerTick, upperTick int64) (sdk.Dec, sdk.Dec, error) {
-	sqrtPriceUpperTick, err := TickToSqrtPrice(sdk.NewInt(upperTick))
+func TicksToPrice(lowerTick, upperTick int64, kAtPriceOne sdk.Int) (sdk.Dec, sdk.Dec, error) {
+	priceUpperTick, err := TickToPrice(sdk.NewInt(upperTick), kAtPriceOne)
 	if err != nil {
 		return sdk.Dec{}, sdk.Dec{}, err
 	}
-	sqrtPriceLowerTick, err := TickToSqrtPrice(sdk.NewInt(lowerTick))
+	priceLowerTick, err := TickToPrice(sdk.NewInt(lowerTick), kAtPriceOne)
 	if err != nil {
 		return sdk.Dec{}, sdk.Dec{}, err
 	}
-	return sqrtPriceLowerTick, sqrtPriceUpperTick, nil
+	return priceLowerTick, priceUpperTick, nil
 }
 
-// TickToSqrtPrice takes the tick index and returns the corresponding sqrt of the price.
-// Returns error if fails to calculate sqrt price. Otherwise, the computed value and nil.
-// TODO: test
-func TickToSqrtPrice(tickIndex sdk.Int) (sqrtPrice sdk.Dec, err error) {
-	// If the tick index is positive, we can use the normal equation to calculate the square root price.
-	// However, if the tick index is negative, since we cannot take negative powers with the sdk,
-	// we need to take one over the original equation in order to make the power positive.
-	var sqrtPriceOsmoMath osmomath.BigDec
-	if tickIndex.GTE(sdk.ZeroInt()) {
-		sqrtPriceOsmoMath, err = TickBase.PowerInteger(tickIndex.Uint64()).ApproxSqrt()
-	} else {
-		sqrtPriceOsmoMath, err = osmomath.OneDec().Quo(TickBase.PowerInteger(tickIndex.Abs().Uint64())).ApproxSqrt()
+// TickToPrice returns the price given the following two arguments:
+// 	- tickIndex: the tick index to calculate the price for
+// 	- kAtPriceOne: the value of k at which the starting price of 1 is set
+//
+// If tickIndex is zero, the function returns sdk.OneDec().
+func TickToPrice(tickIndex, kAtPriceOne sdk.Int) (price sdk.Dec, err error) {
```

Comment:
k at price one should be an int right? (Or perhaps even an `int8`), shouldn't range realistically beyond `[-8, 8]`, and ideally within like `[2, 5]`

Reply:
I see kAtPriceOne as an sdk.Int here. Are you suggesting I change this in the proto to be a primitive int instead?

Reply:
Native int makes sense to me as well